core/src/main/scala/com/mesosphere/package.scala

```diff
@@ -0,0 +1,101 @@
+   *       and thus can chose to built context further on top of the current context OR start fresh.
+   *
+   * The types declared here such as [[KvArgs]] and its value classes can be used through out the project.
+   * Following snippet shows an example of how the [[LoggerTakingImplicit]] can be used with the
+   * abstractions defined here :
+   *
+   * {{{
+   *    // import all the implicit defined here in to scope.
+   *    import com.mesosphere._
+   *
+   *    // Make a log statement with no context.
+   *    // This needs an implicit context but the above import bought an empty context in to scope.
+   *    logger.info("hello world!")
+   *
+   *    // Make a log statement with some context.
+   *    logger.info("hello world!")(kv("key1", "value1"))
+   *
+   *    // Make a log statement with more context.
+   *    val moreContext = kv("key1", "value1").and("key2", "value2").and("key3", "value3")
+   *    log.info("hello world!")(moreContext.and("key3", "key4"))
+   *
+   *    // The above statement will log through whatever backend slf4j was tied to statically.
+   *    // All the key value parameters in context are populated in MDC just before the log statement is made and are
+   *    // deleted immediately after. This happens every time the `logger` object is used to generate a log statement.
+   * }}}
+   */
+
+  type KvArgs = Traversable[(String, Any)]
+
+  implicit case object CanLogKvArgsEv extends CanLog[KvArgs] {
+    override def logMessage(originalMsg: String, a: KvArgs): String = {
+      for (elem <- a) {
+        val (key, value) = elem
+        MDC.put(key, value.toString)
+      }
+      originalMsg
+    }
+
+    override def afterLog(a: KvArgs): Unit = a.foreach { case (key, _) => MDC.remove(key) }
+  }
+
+  implicit val emptyKvArgs: KvArgs = Traversable()
+
+  def kv(kv: (String, Any)): KvArgs = Traversable(kv)
+
+  def kv(key: String, value: Any): KvArgs = kv((key, value))
+
+  implicit class KvArgsOps(val self: KvArgs) extends AnyVal {
+    /**
+     * Helper methods to enable fluent composition of context parameter(s).
+     */
+    def and(kv: KvArgs): KvArgs = self ++ kv
+
+    def and(k: String, v: String): KvArgs = self and kv(k, v)
+  }
+
+}
+
+package mesosphere {
+  /**
+   * Defines `logger` as a value initialized with an underlying `org.slf4j.Logger`
+   * named according to the class into which this trait is mixed.
+   *
+   * Also refer [[com.typesafe.scalalogging.StrictLogging]]
+   */
+  trait ImplicitStrictLogging {
+    protected val logger: LoggerTakingImplicit[KvArgs] =
+      Logger.takingImplicit[KvArgs](getClass)
+  }
+
+  /**
+   * Defines `logger` as a value initialized with an underlying `org.slf4j.Logger`
+   * named according to the class into which this trait is mixed.
+   *
+   * Also refer [[com.typesafe.scalalogging.LazyLogging]]
+   */
+  trait ImplicitLazyLogging {
+    @transient
+    protected val logger: LoggerTakingImplicit[KvArgs] =
```

Comment:
How does it differ from `ImplicitStrictLogging`? When should it be used?

Reply:
Updated code to correctly mark this as `lazy`. I've done what was being done in `StrictLogging` and `LazyLogging` in [scalalogging's Logging.scala](https://github.com/lightbend/scala-logging/blob/master/src/main/scala/com/typesafe/scalalogging/Logging.scala)

I _think_ we may end up using the lazy version in those cases where we know we may/may not need a logger associated with a short lived object (that has no companion object may be?!). I can remove this if you think it's better to add it if/when need arises.